src/SSHLibrary/abstractclient.py

```diff
@@ -1061,15 +1069,19 @@ def __init__(self, command, encoding):
         self._encoding = encoding
         self._shell = None
 
-    def run_in(self, shell):
+    def run_in(self, shell, sudo=False,  sudo_password=None):
         """Runs this command in the given `shell`.
 
         :param shell: A shell in the already open connection.
+
+        :param sudo
+         and
+        :param sudo_password are used for executing commands within a sudo session.
         """
         self._shell = shell
-        self._execute()
+        self._execute(sudo, sudo_password)
```

Comment:
It seems to be that it would be better to already here call different method depending on is sudo used or not. Something like this:

```python
if not sudo:
    self._execute()
else:
    self._execute_with_sudo(sudo_password)
```
This would remove the need for dispatching on the Python and Java sides and probably make the architecture a bit simpler.